python/beeai_framework/backend/vector_store.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2025 © BeeAI a Series of LF Projects, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Any, ClassVar
+
+from beeai_framework.backend.embedding import EmbeddingModel
+from beeai_framework.backend.types import Document, DocumentWithScore
+
+
+class VectorStore(ABC):
+    _integration_registry: ClassVar[dict[str, type[VectorStore]]] = {}
+
+    def __init_subclass__(cls, /, **kwargs: Any) -> None:
+        super().__init_subclass__(**kwargs)
+        cls._integration_registry[cls.integration_name.lower()] = cls
```

Comment:
The problem with this pattern is that the user first needs to import those classes. If the user did not import them, the register would become empty. Use the approach in ChatModel (define a set of providers and do lookups) + use with this dynamic approach (optional, might not be needed).